sm/op.c

```diff
@@ -152,8 +152,8 @@ bool m0_sm_op_tick(struct m0_sm_op *op)
 		}
 		wait = (result & M0_SMOP_WAIT) != 0;
 		M0_ASSERT(ergo(wait, ceo->oe_vec->eo_is_armed(ceo)));
-		if (fail)
-			m0_sm_state_and_rc_set(&op->o_sm, 
+		if (fail || op->o_sm.sm_rc!= 0)
```

Comment:
`fail` check can be removed from all places. `op->o_sm.sm_rc!= 0` check should be sufficient.